### PR TITLE
Test to build against staged dependencies for Maven Central

### PR DESCRIPTION
**Pull request by sewe**
Simplifies the Maven and Gradle builds.

Declared dependencies are based on usage analysis by `mvn dependency:analyze`.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -8,62 +8,37 @@ plugins {
...
+	}
 }
 
 dependencies {
-	implementation fileTree(dir: 'lib', exclude: '*-test-utils.jar')
+	implementation platform("com.teamscale:teamscale-bom:$teamscaleVersion")
+
+	implementation 'com.teamscale:teamscale-check-api'
+	implementation 'com.teamscale:teamscale-commons'
+	implementation 'com.teamscale:teamscale-lib-commons'
```
Comment (DreierF) [inline build.gradle:38]: Those are transitive dependencies of `teamscale-check-api` so there should be no need to specify them explicitly.

Makes it also easier from our side to refactor the packages later on

File: build.gradle [hunk 2]
```diff
@@ -8,62 +8,37 @@ plugins {
...
-	testImplementation fileTree(dir: 'lib', include: '*-test-utils.jar')
-	testImplementation 'org.junit.jupiter:junit-jupiter:5.5.2'
-	testImplementation 'org.junit.jupiter:junit-jupiter-engine:5.5.2'
-	testImplementation 'org.assertj:assertj-core:3.13.2'
-	testImplementation 'nl.jqno.equalsverifier:equalsverifier:3.1.10'
-	testImplementation 'org.junit.platform:junit-platform-launcher:1.5.2'
+	testImplementation 'com.teamscale:teamscale-check-api-test-fixtures'
 }
 
 eclipse {
```
Comment (DreierF) [inline build.gradle:44]: Large parts of this eclipse block are no longer needed AFAICS. I guess the first two ifs can be removed and the other one needs to be adjusted to the new layout

File: build.gradle (hunk 1 above)
Comment (sewe) [reply build.gradle:38]: Yes, they are transitive dependencies of `teamscale-check-api`, **but** they are **also** direct dependencies of `teamscale-custom-check-sample`, as can be seen by running `mvn dependency:analyze`, which determines that the bytecode of `teamscale-custom-check-sample` directly mentions, e.g., `org.conqat.lib.commons.collections.CollectionUtils` (from `teamscale-lib-commons`) or `eu.cqse.check.framework.scanner.ELanguage` (from `teamscale-commons`):

```
[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ custom-checks ---
Unable to process: eu.cqse.check.sample.SampleCheckUsingPhase
[WARNING] Used undeclared dependencies found:
[WARNING]    com.teamscale:teamscale-lib-commons:jar:5.9.0-rc3:compile
[WARNING]    com.teamscale:teamscale-commons:jar:5.9.0-rc3:compile
```

I thus very much prefer in making the dependencies that _do_ exist explicit.

Comment (DreierF) [reply build.gradle]: Ah ok, makes sense.

**Comment by DreierF** [comment]
Importing in IntelliJ works like a charm:
File > Open > Select the folder > Select Open as Gradle (you can also select Maven here) Done

Tested both Gradle and Maven and both load without any errors and tests can be executed in the IDE.
Therefore I would remove the `idea` plugin and adjust the README accordingly.

**Pull request merged**